### PR TITLE
frontend: reorder fields to avoid conflicts and improve readability

### PR DESCRIPTION
**Pull request by bobheadxi**
A few adjustments to alleviate a few pain points:

* Keep high-change fields away from each other to avoid conflicts (such as https://github.com/sourcegraph/deploy-sourcegraph-dogfood-k8s-2/pull/737/files#diff-ec7bc0ca662869dbbc2ef4d656c14002c792de90b497619eacc312397347905d - custom env variables cause perpetual conflicts on image changes because the fields are next to each other
* Scrolling down 50 lines to find the name of a container feels less than ideal

### Checklist

* [ ] Sister [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker) change:
* [ ] All images have a valid tag and SHA256 sum

### CODE REVIEW

**Review by efritz — APPROVED** [review]

File: base/frontend/sourcegraph-frontend.Deployment.yaml [hunk 1]
```diff
@@ -27,7 +27,9 @@ spec:
         deploy: sourcegraph
     spec:
       containers:
-      - args:
+      - name: frontend
+        image: index.docker.io/sourcegraph/frontend:insiders@sha256:b69a210ec21ca10028937dd404467bc84b8da18b31772705199666bc70bc2647
+        args:
```
Comment (efritz) [inline base/frontend/sourcegraph-frontend.Deployment.yaml:32]: - args?

Comment (efritz) [reply base/frontend/sourcegraph-frontend.Deployment.yaml:32]: `-args`?

Comment (efritz) [inline base/frontend/sourcegraph-frontend.Deployment.yaml:32]: `-args?`

**Review by efritz — CHANGES_REQUESTED** [review]

File: base/frontend/sourcegraph-frontend.Deployment.yaml (hunk 1 above)
Comment (bobheadxi) [reply base/frontend/sourcegraph-frontend.Deployment.yaml:32]: `serve` is the arg here (next line)

**Review by efritz — APPROVED** [review]

File: base/frontend/sourcegraph-frontend.Deployment.yaml (hunk 1 above)
Comment (efritz) [reply base/frontend/sourcegraph-frontend.Deployment.yaml:32]: Ugh sorry, the yaml change threw me I thought it was an illegal key for a second. Realized the whole thing was a map.

Comment (bobheadxi) [reply base/frontend/sourcegraph-frontend.Deployment.yaml:32]: git be like that sometimes haha

**Review by davejrt — APPROVED** [review]

**Pull request merged**